### PR TITLE
Upgrade to bitflags 2

### PR DESCRIPTION
**Pull request by djc**
A rebased version of #549 (retaining authorship from the OP).

cc @mrobinson following up on your plan from https://github.com/servo/core-foundation-rs/pull/521#issuecomment-2042258249.

### CODE REVIEW

File: core-text/src/font.rs [hunk 1]
```diff
@@ -1006,9 +1009,41 @@ fn out_of_range_variations() {
...
+            let tag = axis
+                .find(unsafe { kCTFontVariationAxisIdentifierKey })
+                .unwrap();
+            let max = axis
+                .find(unsafe { kCTFontVariationAxisMaximumValueKey })
+                .unwrap()
+                .downcast::<CFNumber>()
+                .unwrap()
+                .to_f64()
+                .unwrap();
```
Comment (mrobinson) [inline core-text/src/font.rs:1030]: It looks like this code is unrelated?

**Review by mrobinson — COMMENTED** [review]
Thanks for the rebase. It looks like this version contains some unrelated code, so I have just rebased the original PR and will merge there.

**Comment by djc** [comment]
It contains code to make the tests pass on macOS 14 (which includes my dev machine). That still seems useful?

**Comment by mrobinson** [comment]
> It contains code to make the tests pass on macOS 14 (which includes my dev machine). That still seems useful?

Yes, it could definitely be useful. Is it related to the bitflags changes though? Would you mind opening a new PR with these changes separating it out from the bitflags work?
